src/plugins/undoRedo/undoRedo.js

```diff
@@ -34,6 +34,15 @@ function UndoRedo(instance) {
     if (!changesLen || ['UndoRedo.undo', 'UndoRedo.redo', 'MergeCells'].includes(source)) {
       return;
     }
+    const hasDifferences = changes.find((change) => {
+      const [,, oldValue, newValue] = change;
+
+      return oldValue !== newValue;
```

Comment:
This won't work correctly (?), when a dataset contains numbers (for example, `data:[[0], [0], [0], [0]]`), with the column not being configured as `numeric`. In that case, editing `0` and pressing <kbd>ENTER</kbd> will result in comparing `0` to `"0"` and marking it as a valid change.

I'm not sure if it should be supported, @swistach, @wojciechczerniak?

Reply:
Will happen if Handsontable was initialized with `data` in different format than `columns` are configured for. Similar problem occurs with validation, where `loadData` allows invalid format but no error will be shown until editor is opened and closed.

We should resolve this issue globally for Handsontable initialization and related issues.